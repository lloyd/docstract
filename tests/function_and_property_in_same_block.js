/** You can't have a single docblock that's both a function
 * AND a property! */

/** @function foo @property bar */
