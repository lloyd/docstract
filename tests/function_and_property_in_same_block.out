{
  "args": [
    "@function and @property may not occur in same documentation block at line 4", 
    []
  ], 
  "exception_type": "<type 'exceptions.RuntimeError'>", 
  "stack": [
    [
      "<module>", 
      "got = ds.extractFromFile(os.path.join(testDir, test + \".js\"))"
    ], 
    [
      "extractFromFile", 
      "data = self.extract(contents)"
    ], 
    [
      "extract", 
      "self._analyzeBlock(block, context, firstBlock, data, lineStart, line)"
    ], 
    [
      "_analyzeBlock", 
      "self._consumeToks(tokens, parseData)"
    ], 
    [
      "_consumeToks", 
      "\"not occur in same documentation block\")"
    ]
  ]
}
